code/modules/mob/living/simple_animal/hostile/mining/goliath.dm

```diff
@@ -195,3 +252,329 @@
+	var/mutable_appearance/draconian_overlay
+
+/mob/living/simple_animal/hostile/asteroid/goliathjuv/subadult
+	growth = 599
+
+/mob/living/simple_animal/hostile/asteroid/goliathjuv/Stat()
+	..()
+	if(statpanel("Status"))
+		stat(null, "Growth: [(growth*100)/1200]%.")
+
+/mob/living/simple_animal/hostile/asteroid/goliathjuv/examine(mob/user)
+	. = ..()
+	if(stat != DEAD)
+		var/list/msgs = list()
+		if(draconian)
+			msgs += "<span class='warning'>It has streaks of magma coursing through its body!</span>"
+		if(key)
+			msgs += "<span class='warning'>It appears to be more aware of its surroundings.</span>"
+		if(tamed == 1)
+			msgs += "<span class='notice'>It appears to be passive.</span>"
+		if(tamed == 2)
+			msgs += "<span class='notice'>It appears to have been tamed.</span>"
+		if(tamed != 2) // No need to let people know if it wants to eat something if it's tamed
+			if(food_wanted == MEAT)
+				msgs += "<span class='notice'>You think you could feed it something...</span>"
+			if(food_wanted == CORE)
+				msgs += "<span class='notice'>It seems to want to eat something refreshing!</span>"
+			if(food_wanted == FLORA)
+				msgs += "<span class='notice'>It seems to want to eat something envigorating!</span>"
+			if(food_wanted == ORGANS)
+				msgs += "<span class='notice'>It seems to want to eat something soft!</span>"
+			if(food_wanted == DIAMOND)
+				msgs += "<span class='notice'>It seems to want to eat something crunchy!</span>"
+		. += msgs.Join("<BR>")
+
+/mob/living/simple_animal/hostile/asteroid/goliathjuv/Life()
+	..()
+	if(stat != DEAD)
+		growth++
+		if(tame_progress >= 1 && tamed != 2) // Lose taming progress with time, unless you've been tamed
+			tame_progress--
+		if(feed_cooldown >= 1) // Lower the cooldown to be fed
+			feed_cooldown--
+		handle_preattack()
+		if(tame_progress <= 599 && tamed != 0) // Become feral if left alone for too long
+			tamed = 0
+			visible_message("<span class='warning'>[src] looks frenzied!</span>")
+			food_wanted = MEAT
+		if(tame_progress >= 600 && tamed == 0) // Become neutral
+			tamed = 1
+			visible_message("<span class='warning'>[src] looks pacified...</span>")
+			reroll_food()
+		if(tame_progress >= 2400 && !picking_candidates && !key) // Become player controlled
+			picking_candidates = TRUE
+			becomeaware()
+		if(tame_progress >= 2400 && key && tamed == 1)
+			to_chat(src, "<span class='biggerdanger'>You have been tamed!</span>")
+			to_chat(src, "<span class='danger'>You have been, by being fed and treated well, introduced to the non-aggressive way of living, but you have also been disowned by the creatures you previously coexisted with.</span>")
+			tamed = 2
+			creator.juvgoliathsspawned--
+			creator = null
+	if(tamed == 2)
+		faction = list("neutral")
+	if(tamed == 1)
+		faction |= "neutral"
+	if(tamed == 0)
+		faction = list("mining")
+	if(stat != DEAD && (getBruteLoss() || getFireLoss())) // Due to being young, their cells multiply at a quicker rate, allowing them to heal
```

Comment:
Have this be handled in the goliath code with a variable of how much it heals. No dupe code please